### PR TITLE
Fix bug in masking of output in tensorflow_backend.rnn(..., unroll=False)

### PR DESCRIPTION
**Pull request by andhus**
### Summary
Fixes https://github.com/keras-team/keras/issues/11472 for tensorflow backend.

### PR Overview
- [y] This PR requires new unit tests [y/n] (make sure tests are included)
- [n] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [y] This PR is backwards compatible [y/n]
- [n] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: keras/backend/tensorflow_backend.py [hunk 1]
```diff
@@ -2921,26 +2921,29 @@ def rnn(step_function, inputs, initial_states,
...
+            parallel_iterations=32,
+            swap_memory=True,
+            maximum_iterations=input_length)
 
         if mask is not None:
             if not states:
                 raise ValueError('No initial states provided! '
                                  'When using masking in an RNN, you should '
-                                 'provide initial states '
-                                 '(and your step function should return '
```
Comment (andhus) [inline keras/backend/tensorflow_backend.py]: Shows the legacy leading up to the bug!

**Comment by fchollet** [comment]
This seems to have a meaningful amount of overlap with your other PR. Could you please merge them to make the diff clearer?

**Comment by andhus** [comment]
Closing in favour of https://github.com/keras-team/keras/pull/11499.
